include/agent.hpp

```diff
@@ -0,0 +1,31 @@
+#include "forecastor.hpp"
+#include <vector>
+
+#define RISK_AVERSION 0.5f
+
+
+typedef struct prediction {
```

Comment:
C++ does not use `typedef` although it is not grammatically wrong.  Use correct C++ class declaration like:

```cpp
struct prediction
{
  double price_term;
  double div_ter;
  double constant;
};
```